SafeExamBrowser.SystemComponents/Registry/Registry.cs

```diff
@@ -104,5 +160,107 @@ private void Timer_Elapsed(object sender, ElapsedEventArgs e)
 				}
 			}
 		}
+
+		/// <summary>
+		/// Parses a keyName and returns the basekey for it.
+		/// It will also store the subkey name in the out parameter.
+		/// If the keyName is not valid, we will return false.
+		/// Does not raise Exceptions.
+		/// Supports shortcuts.
+		/// </summary>
+		// yoinked (and partially modified to follow SEB conventions) private Win32 function: https://stackoverflow.com/a/58547945
+		private bool GetBaseKeyFromKeyName(string keyName, out RegistryKey hiveKey, out string subKeyName)
+		{
+			hiveKey = default;
+			subKeyName = default;
+
+			string basekeyName;
+			int i = keyName.IndexOf('\\');
```

Comment:
Dito (var).

Reply:
`i` should be renamed to `index` or better even `baseKeyLength` or alike, thus indicating what value that variable actually holds.